src/modal/modal_window.ts

```diff
@@ -0,0 +1,46 @@
+import {Component, Output, EventEmitter, ElementRef, DynamicComponentLoader, Input, HostListener} from "angular2/core";
+import {ModalDismissReasons} from './modal_dismiss_reasons';
+
+@Component({
+  selector: 'ngb-modal-window',
+  host: {"class": "modal", "role": "dialog", "tabindex": "-1", "style": "display: block;"},
+  template: `
+        <div [class]="'modal-dialog' + (size ? ' modal-' + size : '')">
+            <div class="modal-content" (click)="stopPropagation($event)">
+                <content-sibiling #content></content-sibiling>                          
+            </div>
+        </div>
+    `
+})
+export class NgbModalWindow {
+  @Input() backdrop = true;
+  @Input() keyboard = true;
+  @Input() size: string;
+
+  @Output('close') closeEvent = new EventEmitter();
+  @Output('dismiss') dismissEvent = new EventEmitter();
+
+  constructor(private _elRef: ElementRef, private _dcl: DynamicComponentLoader) {}
+
+  @HostListener('click')
+  backdropClick() {
```

Comment:
Which name is confusing? `backdropClick`? Can you think of an alternative proposal? `backdropClickHandler`?